wolfcrypt/src/sha.c

```diff
@@ -879,7 +904,7 @@ int wc_ShaFinal(wc_Sha* sha, byte* hash)
 
     XMEMCPY(hash, (byte *)&sha->digest[0], WC_SHA_DIGEST_SIZE);
 
-    (void)InitSha(sha); /* reset state */
+    ret = InitSha(sha); /* reset state */
```

Comment:
Could this be stepping on another error code? Not sure this is the right thing to do. Might let ShaFinal() return a success code, where in fact it failed.

Reply:
oh, that's excellent! thank you! that at least solved a different problem I was working on regarding TLS client/server. Indeed the return code was being stomped on when the HW was busy and needed to fall back to SW. 

Not sure it will fix the current PRB problem that was previously passing. The only change I made was to [AES here](https://github.com/gojimmypi/wolfssl/blob/c18fb50744dbf35bbc49fa5ffe0706182a691334/wolfcrypt/src/aes.c#L1098) after @dgarske added the `!defined(WOLFSSL_SILABS_SE_ACCEL)` upstream. 

The `aes.c` logic I had earlier may have been wrong and had been updated in this PR squash. @dgarske please confirm your SILABS is still working as desired with my changes.